src/ui/marker.js

```diff
@@ -500,20 +511,34 @@ export default class Marker extends Evented {
     }
 
     _calculateRotation(): number {
-        if (this._rotationAlignment === "viewport" || this._rotationAlignment === "auto") {
-            return this._rotation;
-        } if (this._map && this._rotationAlignment === "map") {
-            const pos = this._pos;
-            const map = this._map;
-            if (pos && map && map._usingGlobe()) {
+        const pos = this._pos;
+        const map = this._map;
+        if (!map || !pos) { return 0; }
+        const alignment = this.getRotationAlignment();
+        if (alignment === "map") {
+            if (map._usingGlobe()) {
                 const north = map.project(new LngLat(this._lngLat.lng, this._lngLat.lat + .001));
                 const south = map.project(new LngLat(this._lngLat.lng, this._lngLat.lat - .001));
                 const diff = south.sub(north);
                 return this._rotation + radToDeg(Math.atan2(diff.y, diff.x)) - 90;
             }
-            return this._rotation - this._map.getBearing();
+            return this._rotation - map.getBearing();
+        } else if (this._rotationAlignment === "horizon" && map._usingGlobe()) {
```

Comment:
Similar suggestion as the other comment:
```suggestion
        } else if (this._usingHorizonAlignment()) {
```